vignettes/dAIC-Linear-Regression.Rmd

```diff
@@ -0,0 +1,158 @@
+\end{aligned}
+
+For the $\sigma^2$ component,
+
+
+\begin{aligned}
+    \frac{\partial \ell}{\partial \boldsymbol{\sigma^2}} &= \frac{\partial l^{[1]}}{\partial \boldsymbol{\sigma^2}} + \frac{\partial l^{[2]}}{\partial \boldsymbol{\sigma^2}} + \frac{\partial l^{[3]}}{\partial \boldsymbol{\sigma^2}}; \quad \frac{\partial{l^{[1]}}}{\partial \boldsymbol{\sigma^2}} = 0\\
+        &\Downarrow\\
+    \frac{\partial l^{[2]}}{\partial \boldsymbol{\sigma^2}} &= - \frac{n}{2\sigma^2}; \quad     \frac{\partial l^{[3]}}{\partial \boldsymbol{\sigma^2}} = - \frac{\partial}{\partial \sigma^2} \boldsymbol{\varepsilon}^T\boldsymbol{\varepsilon} / (2 \sigma^2) = \boldsymbol{\varepsilon}^T\boldsymbol{\varepsilon} / (2 (\sigma^2)^2)
+\end{aligned}
+
+Note that the score vector will be zero at the MLE giving,
+
+\begin{aligned}
+    \mathcal U(\widehat{\boldsymbol{\theta}}) = \boldsymbol{0} \Rightarrow \begin{cases} \boldsymbol{X}^T (\boldsymbol{Y} - \boldsymbol{X}\widehat{\boldsymbol{\beta}})/ \widehat{\sigma}^2 \Rightarrow \widehat{\boldsymbol{\beta}} = (\boldsymbol{X}^T\boldsymbol{X})^{-1}\boldsymbol{X}^T\boldsymbol{Y}\\ - \frac{1}{2\widehat{\sigma}^2} (n  - (\boldsymbol{Y} - \boldsymbol{X}\widehat{\boldsymbol{\beta}})^T(\boldsymbol{Y} - \boldsymbol{X}\widehat{\boldsymbol{\beta}})/(\widehat{\sigma}^2)) = 0 \Rightarrow \widehat{\sigma}^2 =  (\boldsymbol{Y} - \widehat{\boldsymbol{Y}})^T(\boldsymbol{Y} - \widehat{\boldsymbol{Y}})/n = \widehat{\boldsymbol{\varepsilon}}^T\widehat{\boldsymbol{\varepsilon}}/n\\
+     \end{cases}
+\end{aligned}
+
+The Hessian matrix is thus computed as,
+
+\begin{aligned}
+    \mathcal{H}(\boldsymbol{\theta}) &= \begin{pmatrix} \frac{\partial^2 \ell}{\partial \boldsymbol{\beta} \partial \boldsymbol{\beta}^T} & \frac{\partial^2 \ell}{\partial \boldsymbol{\beta} \partial \sigma^2} \\ \frac{\partial^2 \ell}{\partial \sigma^2 \partial \boldsymbol{\beta}^T} & \frac{\partial^2 \ell}{\partial \sigma^2 \partial \sigma^2} \end{pmatrix}\\
+        \frac{\partial^2\ell}{\partial\boldsymbol{\beta}\partial\boldsymbol{\beta}^T} &= \frac{\partial}{\partial\boldsymbol{\beta}^T} \boldsymbol{X}^T (\boldsymbol{Y} - \boldsymbol{X}\boldsymbol{\beta}) / \sigma^2 = - \frac{\boldsymbol{X}^T\boldsymbol{X}}{\sigma^2}\\
+        \frac{\partial^2\ell}{\partial\boldsymbol{\beta}\partial\sigma^2} &= \frac{\partial}{\partial\sigma^2} \boldsymbol{X}^T \boldsymbol{\varepsilon}^T / \sigma^2 = - \frac{\boldsymbol{X}^T\boldsymbol{\varepsilon}}{(\sigma^2)^2} \Rightarrow \frac{\partial^2\ell}{\partial\sigma^2\partial\boldsymbol{\beta}^T} = - \frac{\boldsymbol{\varepsilon}^T\boldsymbol{X}}{(\sigma^2)^2}\\
+        \frac{\partial^2\ell}{\partial\sigma^2\partial\sigma^2} &= -\frac{1}{2\sigma^2} \frac{\partial}{\partial\sigma^2} (n - \boldsymbol{\varepsilon}^T \boldsymbol{\varepsilon} / \sigma^2) = \frac{n}{2(\sigma^2)^2} - \frac{\boldsymbol{\varepsilon}^T\boldsymbol{\varepsilon}}{(\sigma^2)^3}
```

Comment:
Where does the middle equality come from? Derivative of a constant w.r.t. to sigma^2 isn't zero?

Reply:
Had that factor outside the partial derivate by mistake. Fixed in [42482f8](https://github.com/Displayr/flipRegression/pull/62/commits/42482f83432b6ed8e656129f5fd5d5c50c0605ce)